src/Stratis.Bitcoin.IntegrationTests.Common/EnvironmentMockUpHelpers/NodeBuilder.cs

```diff
@@ -200,18 +132,15 @@ public CoreNode CloneStratisNode(CoreNode cloneNode)
         /// <param name="protocolVersion">Use <see cref="ProtocolVersion.PROTOCOL_VERSION"/> for BTC PoW-like networks and <see cref="ProtocolVersion.ALT_PROTOCOL_VERSION"/> for Stratis PoS-like networks.</param>
         /// <param name="configFileName">The name for the node's configuration file.</param>
         /// <param name="agent">A user agent string to distinguish different node versions from each other.</param>
-        public CoreNode CreateCustomNode(bool start, Action<IFullNodeBuilder> callback, Network network, ProtocolVersion protocolVersion = ProtocolVersion.PROTOCOL_VERSION, IEnumerable<string> args = null, string agent = "Custom")
+        public CoreNode CreateCustomNode(bool start, Action<IFullNodeBuilder> callback, Network network, ProtocolVersion protocolVersion = ProtocolVersion.PROTOCOL_VERSION, string agent = "Custom", NodeConfigParameters configParameters = null)
         {
-            var argsList = args as List<string> ?? args?.ToList() ?? new List<string>();
-
+            configParameters = configParameters ?? new NodeConfigParameters();
             string configFileName = "custom.conf";
-            if (!argsList.Any(a => a.StartsWith("-conf="))) argsList.Add($"-conf={configFileName}");
-            else configFileName = argsList.First(a => a.StartsWith("-conf=")).Replace("-conf=", "");
-
-            string dataDir = this.GetNextDataFolderName(agent);
-            if (!argsList.Any(a => a.StartsWith("-datadir="))) argsList.Add($"-datadir={dataDir}");
+            configParameters.SetDefaultValueIfUndefined("conf", configFileName);
+            configParameters.SetDefaultValueIfUndefined("datadir", this.GetNextDataFolderName(agent));
 
-            return CreateNode(new CustomNodeRunner(dataDir, callback, network, protocolVersion, argsList, agent), start, configFileName);
+            configParameters?.ToList().ForEach(p => this.ConfigParameters[p.Key] = p.Value);
+            return CreateNode(new CustomNodeRunner(this.GetNextDataFolderName(agent), callback, network, protocolVersion, configParameters, agent), start, configFileName);
```

Comment:
If you don't use a `datadir` variable, won't a new folder be created?

Reply:
not sure what you mean here ... There is a use of datadir (there was even 2, which was a mistake, now fixed)